src/Semigroupoids/Do.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE CPP #-}
+
+#if __GLASGOW_HASKELL__ >= 702
+{-# LANGUAGE Safe #-}
+#endif
+
+#if __GLASGOW_HASKELL__ == 708
+{-# OPTIONS_GHC -fno-warn-amp #-}
+#endif
+
+{-|
+
+This module re-exports operators from 'Data.Functor.Apply' and 'Data.Functor.Bind', but under the same
+names as their 'Applicative' and 'Monad' counterparts. This makes it convenient
+to use do-notation on a type that is a 'Bind' but not a monad (or an 'Apply'
+but not an 'Applicative' with @ApplicativeDo@), either using the
+@QualifiedDo@ extension or the more traditional @RebindableSyntax@.
+
+@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE QualifiedDo #-}
+
+foo :: Apply f => f a -> f b -> f (a, b)
+foo as bs = Semi.do
+  a <- as
+  b <- bs
+  pure (a, b)
+
+
+bar :: Bind m => (a -> b -> m c) -> m a -> m b -> m c
+bar f as bs = Semi.do
+  a <- as
+  b <- bs
+  f a b
+@
+
+-}
+module Semigroupoids.Do
+  ( fmap
+  , (<*)
+  , (*>)
+  , (<*>)
+  , (>>)
+  , (>>=)
+  , join
+  , pure
+  , return
+  )
+where
+
+#if !MIN_VERSION_base(4,8,0)
+import Control.Applicative (pure)
+#endif
+import Data.Functor.Apply (Apply, (<.), (.>), (<.>))
+import Data.Functor.Bind (Bind, (>>-), join)
+import Prelude ()
+
+-- | /Since: 5.3.6/
```

Comment:
For consistency with the rest of the package, I'd just write this (and the remaining annotations) using the modern syntax:

```suggestion
-- | @since 5.3.6
```

It'll look a bit funny on very old Haddocks, but I'm not especially worried about that.